File: ecommerce-api-legacy/src/models/userModel.js
const { get, run } = require('../config/database');

class UserModel {
  constructor(db) { this.db = db; }
  findByEmail(email) { return get(this.db, 'SELECT id FROM users WHERE email = ?', [email]); }
  create({ name, email, password }) { return run(this.db, 'INSERT INTO users (name, email, pass) VALUES (?, ?, ?)', [name, email, password]); }
  delete(userId) { return run(this.db, 'DELETE FROM users WHERE id = ?', [userId]); }
}

module.exports = UserModel;

File: code-smells-project/src/models/user_model.py
from src.config.database import get_db

def public_user(row, include_password=False):
    if not row:
        return None
    data = dict(row)
    if not include_password:
        data.pop('senha', None)
    return data

def list_users():
    return [public_user(row) for row in get_db().execute('SELECT * FROM usuarios').fetchall()]

def get_user(user_id):
    return public_user(get_db().execute('SELECT * FROM usuarios WHERE id = ?', (user_id,)).fetchone())

def authenticate(email, senha):
    row = get_db().execute('SELECT id, nome, email, tipo FROM usuarios WHERE email = ? AND senha = ?', (email, senha)).fetchone()
    return dict(row) if row else None

def create_user(nome, email, senha, tipo='cliente'):
    cursor = get_db().execute('INSERT INTO usuarios (nome, email, senha, tipo) VALUES (?, ?, ?, ?)', (nome, email, senha, tipo))
    get_db().commit()
    return cursor.lastrowid
